drive_test.go

```diff
@@ -243,3 +246,134 @@ func TestListDriveChildrenInvalid(t *testing.T) {
 	}
 }
+
+func TestDriveSearchFiles(t *testing.T) {
+	setup()
+	defer teardown()
+
+	t.Run("when successful response", func(t *testing.T) {
+		mux.HandleFunc("/drive/search(q='te%st')", fileWrapperHandler("fixtures/drive.search.valid.json", 200))
+
+		wantItems := &Items{
+			Collection: []*Item{
+				{
+					ID:   "res_1",
+					Name: "Test numbers.xlsx",
+					CreatedBy: &IdentitySet{
+						User: &Identity{
+							DisplayName: "Developer",
+						},
+					},
+					LastModifiedBy: &IdentitySet{
+						User: &Identity{
+							DisplayName: "Developer",
+						},
+					},
+					CreatedDateTime:      time.Date(2019, 9, 23, 14, 20, 38, 0, time.UTC),
+					LastModifiedDateTime: time.Date(2019, 9, 23, 14, 21, 51, 0, time.UTC),
+					Size:                 8426,
+					ParentReference: &ItemReference{
+						DriveID: "b!drive1",
+						ID:      "ref_1",
+					},
+					WebURL: "https://test-my.sharepoint.com/personal/test_onmicrosoft_com/_layouts/15/Doc.aspx?sourcedoc=%7res_1%7D&file=Test%20numbers.xlsx&action=default&mobileredirect=true&DefaultItemOpen=1",
+					File:   newFileFacet("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", nil),
+				},
+				{
+					ID:   "res_2",
+					Name: "Test numbers 2.xlsx",
+					CreatedBy: &IdentitySet{
+						User: &Identity{
+							DisplayName: "Developer",
+						},
+					},
+					LastModifiedBy: &IdentitySet{
+						User: &Identity{
+							DisplayName: "Developer",
+						},
+					},
+					CreatedDateTime:      time.Date(2019, 9, 23, 14, 20, 38, 0, time.UTC),
+					LastModifiedDateTime: time.Date(2019, 9, 23, 14, 21, 51, 0, time.UTC),
+					Size:                 5426,
+					ParentReference: &ItemReference{
+						DriveID: "b!drive1",
+						ID:      "ref_1",
+					},
+					WebURL: "https://test-my.sharepoint.com/personal/test_onmicrosoft_com/_layouts/15/Doc.aspx?sourcedoc=%7res_1%7D&file=Test%20numbers%202.xlsx&action=default&mobileredirect=true&DefaultItemOpen=1",
+					File:   newFileFacet("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", nil),
+				},
+			},
+		}
+
+		items, _, err := oneDrive.Drives.Search(context.Background(), "te%st")
+		if err != nil {
+			t.Fatal("unexpected error", err)
+		}
+
+		wantCount := 2
+		if len(items.Collection) != wantCount {
+			t.Errorf("unexpected item count want %d but got %d", wantCount, len(items.Collection))
+		}
+
+		if diff := cmp.Diff(wantItems, items); diff != "" {
+			t.Error("unexpected diff", diff)
+		}
+	})
+
+	t.Run("dangerous characters sanitized from query", func(t *testing.T) {
+		specs := []struct {
+			in  string
+			out string
+		}{
+			{"test*", "test"},
+			{"test<tom", "testtom"},
+			{">", ""},
+			{"test's & skill's", "tests & skills"},
+			{"some_path\testfile", "some_pathtestfile"},
+			{"some_path/testfile", "some_pathtestfile"},
+			{"test?", "test"},
+		}
+
+		for _, spec := range specs {
+			mux.HandleFunc("/drive/search(q='"+spec.out+"')", fileWrapperHandler("fixtures/drive.search.valid.json", 200))
```

Comment:
🏄 I'd be tempted to make this handler be at `/drive`, and have it record the `r.URL.Path` it saw, and then the assertions could be `expected: path, got: other-path`, instead of a failing case giving a 404.